test/monitoring_console/monitoring_console_test.go

```diff
@@ -0,0 +1,532 @@
+			Expect(err).To(Succeed(), "Unable to deploy search head cluster")
+			// Ensure search head cluster go to Ready phase
+			shc := &enterprisev1.SearchHeadCluster{}
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), shc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for search head cluster instance status to be ready", "instance", shc.ObjectMeta.Name, "Phase", shc.Status.Phase)
+				testenv.DumpGetPods(testenvInstance.GetName())
+				return shc.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// In a steady state, we should stay in Ready and not flip-flop around
+			Consistently(func() splcommon.Phase {
+				_ = deployment.GetInstance(deployment.GetName(), shc)
+				return shc.Status.Phase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Wait for 1 MC pod in namespace
+			Eventually(func() int {
+				count := testenv.GetMCPodCount(testenvInstance.GetName())
+				return count
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(1))
+
+			// Check Monitoring Console Pod is in Ready State
+			Eventually(func() bool {
+				check := testenv.CheckMCPodReady(testenvInstance.GetName())
+				return check
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(true))
+
+			// Check Monitoring console is configured with all search head instances in namespace
+			peerList := testenv.GetConfiguredPeers(testenvInstance.GetName())
+			found := make(map[string]bool)
+			testenvInstance.Log.Info("Peer List", "instance", peerList)
+			for i := 0; i < 3; i++ {
+				podName := "splunk-" + deployment.GetName() + "-search-head-" + strconv.Itoa(i)
+				found[podName] = false
+				for _, peer := range peerList {
+					if strings.Contains(peer, podName) {
+						testenvInstance.Log.Info("Check Peer matches serach head pod", "Search Head Pod", podName, "Peer in peer list", peer)
+						found[podName] = true
+						break
+					}
+				}
+			}
+			allSearchHeadsConfigured := true
+			for _, key := range found {
+				if !key {
+					allSearchHeadsConfigured = false
+					break
+				}
+			}
+			Expect(allSearchHeadsConfigured).To(Equal(true))
+
+			// Scale Search Head Cluster
+			testenvInstance.Log.Info("Scaling search head cluster")
+			_, err = exec.Command("kubectl", "scale", "shc", "-n", testenvInstance.GetName(), deployment.GetName(), "--replicas=4").Output()
+			Expect(err).To(Succeed(), "Failed to scale search head cluster")
+
+			// Ensure search head cluster go to ScalingUp phase
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), shc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for search head cluster instance status to be Scaling Up", "instance", shc.ObjectMeta.Name, "Phase", shc.Status.Phase)
```

Comment:
%s/"Waiting for search head cluster instance status"/"Waiting for search head cluster status"